gateway/dataimpl/server_v1/errorhandler.go

```diff
@@ -590,7 +595,12 @@ func (e ErrorHandler) NewGenericStatus(err error) *status.Status {
 	return e.NewUnknownStatus(err)
 }
 
-func (e ErrorHandler) NewKeyTooLongStatus(key string) *status.Status {
-	st := status.New(codes.InvalidArgument, fmt.Sprintf("Document key '%s' is too long", key))
+func (e ErrorHandler) NewInvalidKeyLengthStatus(key string) *status.Status {
```

Comment:
We should avoid trying to be overly specific with our error builders, and instead have a particular error capture one specific case that has occured, either by consolidating the error message or having two methods for each kind of error.  We otherwise risk having multiple places that the error condition must be validated (for instance, in this case, the len(key)<1 check is happening both in CNGs handlers, and then again in the error builder.  In this case, I think it's probably best to just adjust the message to indicate "Document key ... must be between 1 and 250 characters".